ffs/filcold/filcold.go

```diff
@@ -260,15 +269,55 @@ func (fc *FilCold) WaitForDeals(ctx context.Context, c cid.Cid, proposals []cid.
 		res = append(res, *v)
 	}
 	fc.l.Log(ctx, c, "All deals reached final state.")
+
 	return res, errors, nil
 }
 
-func (fc *FilCold) getCidSize(ctx context.Context, c cid.Cid) (uint64, error) {
-	s, err := fc.ipfs.Object().Stat(ctx, path.IpfsPath(c))
+// estimatePieceSize estimates the size of the Piece that will be built by the filecoin client
+// when making the deal. This calculation should consider the unique node sizes of the DAG, and
+// padding. It's important to not underestimate the size since that would lead to deal rejection
+// since the miner won't accept the further calculated PricePerEpoch.
+func (fc *FilCold) calculatePieceSize(ctx context.Context, c cid.Cid) (uint64, error) {
```

Comment:
So the thing here is calculating the CAR file serialization without making the CAR, plus considering the padding that is needed for pieces in Filecoin. 

Not making the CAR makes this scale for bigger files. So this thing is bound by the # of nodes in the DAG, which presumably should be ~_log(size of data)_ API calls (assuming the UnixFS DAG is balanced). Work isn't related with byte size since no data is fetched in the apis.

Reply:
I played making some tricks about returning the calculated size minus 1 to let the deal fail, and it seems its calculating exactly the minimum price. It may need more data sizes tests, but looks quite promising. We'll discover if this isn't that correct, because the deal proposal will fail for some use case. So, quite easy to spot.